src/hotspot/share/opto/vtransform.hpp

```diff
@@ -522,6 +522,22 @@ class VTransformStoreVectorNode : public VTransformMemVectorNode {
   NOT_PRODUCT(virtual const char* name() const override { return "StoreVector"; };)
 };
 
+class VTransformCastVectorNode : public VTransformNode {
+private:
+  uint _vlen;
+  BasicType _from_bt;
+  BasicType _to_bt;
+
+public:
+  // req = 2 -> [ctrl, input]
+  VTransformCastVectorNode(VTransform& vtransform, int vlen, BasicType from_bt, BasicType to_bt) : VTransformNode(vtransform, 2),
+    _vlen(vlen), _from_bt(from_bt), _to_bt(to_bt) {}
+  virtual VTransformApplyResult apply(const VLoopAnalyzer& vloop_analyzer,
+                                      const GrowableArray<Node*>& vnode_idx_to_transformed_node) const override;
+  NOT_PRODUCT(virtual const char* name() const override { return "Cast"; };)
```

Comment:
```suggestion
  NOT_PRODUCT(virtual const char* name() const override { return "CastVector"; };)
```

Reply:
I know the node is not inheriting from `VTransformVectorNode`, but we can make that happen with a future refactoring I'm already working on.